code/datums/storage/storage.dm

```diff
@@ -0,0 +1,1027 @@
+	///storage UI
+	var/atom/movable/screen/storage/boxes = null
+	///storage UI
+	var/atom/movable/screen/storage/storage_start = null
+	///storage UI
+	var/atom/movable/screen/storage/storage_continue = null
+	///storage UI
+	var/atom/movable/screen/storage/storage_end = null
+	///storage UI
+	var/atom/movable/screen/storage/stored_start = null
+	///storage UI
+	var/atom/movable/screen/storage/stored_continue = null
+	///storage UI
+	var/atom/movable/screen/storage/stored_end = null
+	///storage UI
+	var/atom/movable/screen/close/closer = null
+
+	///whether our storage box on hud changes color when full.
+	var/show_storage_fullness = TRUE
+	///Set this to make it possible to use this item in an inverse way, so you can have the item in your hand and click items on the floor to pick them up.
+	var/use_to_pickup
+	///Set this to make the storage item group contents of the same type and display them as a number.
+	var/display_contents_with_number
+	///Set this variable to allow the object to have the 'empty' verb, which dumps all the contents on the floor.
+	var/allow_quick_empty
+	///Set this variable to allow the object to have the 'toggle mode' verb, which quickly collects all items from a tile.
+	var/allow_quick_gather
+	///whether this object can change its drawing method
+	var/allow_drawing_method
+	///FALSE = will open the inventory if you click on the storage container, TRUE = will draw from the inventory if you click on the storage container
+	var/draw_mode = FALSE
+	///FALSE = pick one at a time, TRUE = pick all on tile
+	var/collection_mode = TRUE
+	///BubbleWrap - if set, can be folded (when empty) into a sheet of cardboard
+	var/foldable = null
+	///sound played when used. null for no sound.
+	var/use_sound = "rustle"
+	///Has it been opened before?
+	var/opened = 0
+	///list of mobs currently seeing the storage's contents
+	var/list/content_watchers = list()
+	///How long does it take to put items into or out of this, in ticks
+	var/access_delay = 0
+	///What item do you use to tactical refill this
+	var/list/obj/item/refill_types
+	///What sound gets played when the item is tactical refilled
+	var/refill_sound = null
+	///the item left behind when our parent is destroyed
+	var/trash_item = null
+	///flags for special behaviours
+	var/storage_flags = NONE
+
+	//---- Holster vars
+	///the sound produced when the special item is drawn
+	var/draw_sound = null
+	///the sound produced when the special item is sheathed
+	var/sheathe_sound = null
+	///the snowflake item(s) that will update the sprite.
+	var/list/holsterable_allowed = list()
+	///records the specific special item currently in the holster
+	var/obj/holstered_item = null
+	///Image that get's underlayed under the sprite of the holster
+	var/image/holstered_item_underlay
+
+
+/datum/storage/New(atom/parent)
+	. = ..()
+	if(!istype(parent))
+		stack_trace("Storage datum ([type]) created without a [isnull(parent) ? "null parent" : "invalid parent ([parent.type])"]!")
+		qdel(src)
```

Comment:
why a qdel ?

Reply:
This was ported from TG, I assume it's to prevent harddels or some weird instance of storage existing *somewhere* but without a parent